hs-abci-sdk/src/Tendermint/SDK/BaseApp/Events.hs

```diff
@@ -121,6 +125,35 @@ emit
   -> Sem r ()
 emit e = output $ makeEvent e
 
+-- | Special event wrapper to add contextual event_type info
+newtype ContextEvent t = ContextEvent t
+instance (A.ToJSON a, ToEvent a) => A.ToJSON (ContextEvent a) where
+  toJSON (ContextEvent a) =
+    A.object [ "event_type" A..= makeEventType (Proxy :: Proxy a)
+             , "event" A..= A.toJSON a
+             ]
+instance Log.Select a => Log.Select (ContextEvent a) where
+  select v (ContextEvent a) = Log.select v a
+
+logEvent
+  :: forall e r.
+     (A.ToJSON e, ToEvent e, Log.Select e)
+  => Member Log.Logger r
+  => e
+  -> Sem r ()
+logEvent event = Log.addContext (ContextEvent event) $
+  Log.log Log.Info (cs $ makeEventType (Proxy :: Proxy e))
+
+emitAndLogEvent
```

Comment:
I don't think we should put this one in the library, it's simple enough to call both functions